### PR TITLE
Add client endpoints docs

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

This exposes endpoints in the docs; I did not expose `client.py` because hightouch credentials is the entry point for getting a client.

image
image

## Relevant Issue(s)

Closes https://github.com/PrefectHQ/prefect-hightouch/issues/6

## Checklist
- [ ] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-hightouch/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_hightouch/api_client/api/default/get_destination.py [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+
+The functions are described below:
+
+- `asyncio`: Like sync but async instead of blocking
+
+- `asyncio_detailed`: Like sync_detailed but async instead of blocking
+
+- `sync`: Blocking request that returns parsed data (if successful) or None
+
+- `sync_detailed`: Blocking request that always returns a Request, optionally with parsed set if the request was successful.
```
Comment (desertaxle) [inline prefect_hightouch/api_client/api/default/get_destination.py]: Leaning towards being extra explicit is useful for these modules I think.

Tangential question: where do the templates for these generated files live?
```suggestion
- `asyncio`: Non-blocking request that returns parsed data (if successful) or None. Any call must be awaited.

- `asyncio_detailed`: Non-blocking request that always returns a Request, optionally with parsed set if the request was successful. Any call must be awaited.

- `sync`: Blocking request that returns parsed data (if successful) or None.

- `sync_detailed`: Blocking request that always returns a Request, optionally with parsed set if the request was successful.
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: docs/endpoints/trigger_run_custom.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+::: prefect_hightouch.api_client.api.default.trigger_run_custom
+    options:
+      filters: ["!^__[^__]", "!Response"]
```
Comment (zzstoatzz) [inline docs/endpoints/trigger_run_custom.md]: is there a way we could set this filter in the mkdocs.yaml if its consistent among all these default endpoints?

File: prefect_hightouch/credentials/generated.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ class HightouchCredentials(Block):
     """
 
     _block_type_name = "Hightouch Credentials"
-    # _logo_url = "<LOGO_URL_HERE>"  # noqa
+    # _logo_url = "<UPDATE _logo_url IN __init__.py>"  # noqa
```
Comment (zzstoatzz) [inline prefect_hightouch/credentials/generated.py]: I'm probably just lacking context with the generator, but why is this necessary?

Comment (ahuang11) [reply prefect_hightouch/credentials/generated.py]: To not override user input

**Review by desertaxle — APPROVED** [review]
